src/state.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
The contracts charm includes some unit test for this State object, I think it's worth copying them here. See https://github.com/canonical/contracts/blob/main/charm/tests/unit/test_charm.py#L358